pkg/events/event_system.go

```diff
@@ -38,20 +42,57 @@ type EventSystem interface {
 }
 
 type EventSystemImpl struct {
-	Store     *EventStore // storing eventChannel
-	publisher *EventPublisher
+	eventSystemId string
+	Store         *EventStore // storing eventChannel
+	publisher     *EventPublisher
 
 	channel chan *si.EventRecord // channelling input eventChannel
 	stop    chan bool            // whether the service is stop
 	stopped bool
 
-	sync.Mutex
+	trackingEnabled bool
```

Comment:
Is tracking the entire event system? I suggest a name like "eventsEnabled" or "eventSystemEnabled".

Reply:
I think the existing name is fine. We actually need to leave the existing event system always ON as we will be needing those events in the shim for other purposes (better reporting of predicate checks to K8s for example). Therefore, trackingEnabled seems to fit -- we are controlling only whether we store events for retrieval via API.